src/pyedb/grpc/database/components.py

```diff
@@ -276,20 +276,26 @@ def refresh_components(self):
         self._others = {}
         for i in self._pedb.layout.groups:
             self._cmp[i.name] = i
-            if i.type == "resistor":
-                self._res[i.name] = i
-            elif i.type == "capacitor":
-                self._cap[i.name] = i
-            elif i.type == "inductor":
-                self._ind[i.name] = i
-            elif i.type == "ic":
-                self._ics[i.name] = i
-            elif i.type == "io":
-                self._ios[i.name] = i
-            elif i.type == "other":
+            try:
```

Comment:
Is it possible to avoid using try-except?

Reply:
No sorry because the grpc internal api crashes. We spoke with @SMoraisAnsys about this and we will add a codacy exception to prevent warnings. I am in with dev to improve this behvior, that was actually done on design. While DotNet never crashes without throwing errors the grpc is the opposite. That's btw how I saw that DotNet tests were not tested properly and even failing silently.